### PR TITLE
Fix gitlab CI

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

Once the `substrate-subxt` **v0.1 branch** is updated, I will revert `cli/Cargo.toml`.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -88,12 +88,6 @@ cargo-test-af:
   script:
   - cargo test --verbose --all --all-features
 
-cargo-build-nf:
-  stage:                           check-workspace
-  <<:                              *docker-env
-  script:
-  - cargo build --verbose --all --no-default-features --release --target=wasm32-unknown-unknown
-
```
Comment (Robbepop) [inline .gitlab-ci.yml]: So you fix it by removing it?

File: cli/Cargo.toml [hunk 2]
```diff
@@ -22,13 +22,13 @@ itertools = "0.8"
...
 futures = "0.1.28"
-jsonrpc-core-client = { version = "13.0", features = ["ws"] }
+jsonrpc-core-client = { version = "14.0", features = ["ws"] }
 zip = { version = "0.5", default-features = false }
 pwasm-utils = "0.11.0"
 parity-wasm = "0.40.2"
 cargo_metadata = "0.8.2"
 substrate-primitives = { git = "https://github.com/paritytech/substrate/", package = "substrate-primitives" }
-subxt = { git = "https://github.com/paritytech/substrate-subxt/", branch = "v0.1", package = "substrate-subxt" }
+subxt = { git = "https://github.com/paritytech/substrate-subxt/", branch = "master", package = "substrate-subxt" }
```
Comment (Robbepop) [inline cli/Cargo.toml]: Not sure if we really want to be using `master` because that sounds rather unstable to me.

Comment (koushiro) [reply cli/Cargo.toml]: Once the `substrate-subxt` v0.1 branch is updated, I will revert it.

File: .gitlab-ci.yml (hunk 1 above)
Comment (koushiro) [reply .gitlab-ci.yml]: `net2` will build error on target `wasm32-unknown-unknwon`

File: cli/Cargo.toml (hunk 2 above)
Comment (Robbepop) [reply cli/Cargo.toml]: @ascjones comment needed

**Comment by koushiro** [comment]
Maybe we need to add `rust-toolchain`, otherwise it will make different results for `rustfmt` and `clippy`.

**Comment by Robbepop** [comment]
> Maybe we need to add `rust-toolchain`, otherwise it will make different results for `rustfmt` and `clippy`.

We just got rid of `rust-toolchain` but maybe you mean only on the CI level.

File: cli/Cargo.toml (hunk 2 above)
Comment (ascjones) [reply cli/Cargo.toml]: Branch `v0.1` updated

**Comment by koushiro** [comment]
ICE occurs when running `clippy`, maybe we should disable `clippy` check temporarily. What do you think? @Robbepop

**Comment by koushiro** [comment]
PTAL again @Robbepop

**Comment by koushiro** [comment]
@Robbepop Please fix tests in `ink_lang2_macro`

**Comment by Robbepop** [comment]
Closed since other PR already improved CI substantially. Also no feedback. Closed.

**Pull request closed without merging**